src/store/ModuleInfoStore.js

```diff
@@ -1,192 +1,197 @@
-      .then(readmeEncoded => {
-        const readmeDecoded = atob(readmeEncoded.content);
-        const readmeHtml = marked(readmeDecoded);
-
-        setState({
-          type: READ_ME_CHANGED,
-          payload: {
-            readme: readmeHtml
-          }
-        });
-      })
-      .catch(err => console.log(err));
-  };
-
-  // Helper methods used to help the state changing methods above
-
-  const _getRepoNameAndSundays = clickEvent => {
-    // it is a promise because it is being used in the getReadme function and that one won't wait
-    let repoName = null;
-    const target = clickEvent.target;
-    const dataset = target.dataset;
-    const {
-      start,
-      end,
-      repo,
-      group_id,
-      group,
-      running_module_id,
-      duration
-    } = dataset;
-
-    // if the selected element doesn't have a repo
-    if (!repo) {
-      repoName = noRepoAlternative;
-      setState({});
-    } else {
-      repoName = repo;
+
+    const _getSundays = (startString, endString) => {
+        // note: not passing just a string because moment is throwing a warning about the format of the string
+        const start = moment(new Date(startString))
+        const end = moment(new Date(endString))
+        const allSundays = []
+        while (start.day(0).isBefore(end)) {
+            allSundays.push(start.clone().format('YYYY/MM/DD'))
+            start.add(1, 'weeks')
+        }
+        return allSundays
+    }
+
+    const defaultReadme = async defaultRepo => {
+        try {
+            const res = await fetch(`${BASE_URL}/${defaultRepo}/readme`)
+            const readmeEncoded = await res.json()
+            const readmeDecoded = atob(readmeEncoded.content)
+            const readmeHtml = marked(readmeDecoded)
+
+            setState({
+                type: READ_ME_CHANGED,
+                payload: {
+                    readme: readmeHtml,
+                    repoName: defaultRepo
+                }
+            })
+            
+        } catch(error){
+            console.log(error)
+        }
     }
 
-    const mergedData = {
-      type: REPO_NAME_CHANGED,
-      payload: {
-        repoName,
-        group_id,
-        running_module_id,
-        duration,
-        start,
-        end,
-        group
-      }
-    };
-    setState(mergedData);
-  };
-
-  const _getSundays = (startString, endString) => {
-    // note: not passing just a string because moment is throwing a warning about the format of the string
-    const start = moment(new Date(startString));
-    const end = moment(new Date(endString));
-    const allSundays = [];
-    while (start.day(0).isBefore(end)) {
-      allSundays.push(start.clone().format('YYYY/MM/DD'));
-      start.add(1, 'weeks');
+    return {
+        subscribe,
+        unsubscribe,
+        getState,
+        setState,
+        getReadme,
+        getHistory,
+        defaultReadme
     }
-    return allSundays;
-  };
-
-  const defaultReadme = defaultRepo => {
```

Comment:
in my local machine this async await but i don't understand why it is different here